sql/src/main/java/io/crate/operation/tablefunctions/GenerateSeriesFunction.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.tablefunctions;
+
+import io.crate.analyze.WhereClause;
+import io.crate.data.Bucket;
+import io.crate.data.Input;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.metadata.*;
+import io.crate.metadata.table.StaticTableInfo;
+import io.crate.metadata.table.TableInfo;
+import io.crate.metadata.tablefunctions.TableFunctionImplementation;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.cluster.service.ClusterService;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+public class GenerateSeriesFunction {
+
+    private static final String NAME = "generate_series";
+    private static final TableIdent TABLE_IDENT = new TableIdent(null, NAME);
+
+    static class GenerateSeriesFunctionImplementation implements TableFunctionImplementation {
+
+        private final FunctionInfo info;
+
+        private GenerateSeriesFunctionImplementation(FunctionInfo info) {
+            this.info = info;
+        }
+
+        @Override
+        public FunctionInfo info() {
+            return info;
+        }
+
+        private static Long[] extractValues(Collection<? extends Input> arguments) {
+            assert arguments.size() >= 2 && arguments.size() <= 3 : NAME + " only accepts 2-3 arguments";
+
+            Long[] values = new Long[arguments.size()];
+            int idx = 0;
+
+            for (Input argument : arguments) {
+                Number value = (Number) argument.value();
+                values[idx++] = value.longValue();
+            }
+
+            return values;
+        }
+
+        @Override
+        public Bucket execute(Collection<? extends Input> arguments) {
+            final Long[] values = extractValues(arguments);
+
+            final long start = values[0];
```

Comment:
The argument values could be extracted directly without first building up another array